src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1076,6 +1076,13 @@ static inline bool notification_match(reservation_t& res,
   return true;
 }
 
+static inline uint32_t compute_target_shard(const DoutPrefixProvider* dpp, const std::string& bucket_name, const std::string& object_key, const uint32_t num_shards){
+  std::hash<std::string> hash_fn; 
+  size_t hash = hash_fn(bucket_name + ":" + object_key); 
```

Comment:
nit: plus operator on strings is inefficient. please use `fmt::format()` instead